zenoh-flow/src/model/descriptor/node/mod.rs

```diff
@@ -154,20 +168,45 @@ impl NodeDescriptor {
         self,
         global_configuration: Option<Configuration>,
     ) -> Result<SourceDescriptor> {
-        let descriptor = try_load_descriptor(&self.descriptor).await?;
+        log::trace!("[Descriptor] Loading Source {}", self.id);
 
-        log::trace!("Loading source {}", self.descriptor);
+        match parse_uri(&self.descriptor)? {
+            URIStruct::File(path) => {
+                let descriptor = try_load_descriptor_from_file(path).await?;
 
-        match SourceDescriptor::from_yaml(&descriptor) {
-            Ok(mut desc) => {
-                desc.id = self.id;
-                desc.configuration = global_configuration.merge_overwrite(desc.configuration);
-                Ok(desc)
-            }
-            Err(e) => {
-                log::warn!("Unable to flatten {}, error {}", self.id, e);
-                Err(e)
+                match SourceDescriptor::from_yaml(&descriptor) {
+                    Ok(mut desc) => {
+                        desc.id = self.id;
+                        desc.configuration =
+                            global_configuration.merge_overwrite(desc.configuration);
+                        Ok(desc)
+                    }
+                    Err(e) => {
+                        log::warn!("Unable to read descriptor {}, error {}", self.id, e);
+                        Err(e)
+                    }
+                }
             }
+            URIStruct::Builtin(mw, kind) => match mw {
+                Middleware::Zenoh => match kind {
+                    NodeKind::Operator => bail!(
+                        ErrorKind::Unimplemented,
+                        "Cannot load an operator as source"
+                    ),
+                    NodeKind::Source => match &self.configuration {
+                        Some(configuration) => get_zenoh_source_descriptor(configuration),
+                        None => {
+                            bail!(
+                                ErrorKind::MissingConfiguration,
+                                "Builtin Zenoh Sink needs a configuration!"
+                            )
+                        }
+                    },
+                    NodeKind::Sink => {
+                        bail!(ErrorKind::Unimplemented, "Cannot load an sink as source")
+                    }
+                },
+            },
```

Comment:
It feels like `kind` is redundant and introduces possibilities for errors (as shown by the 3 branches of the`match`).
If a builtin node is declared under the `sources` section then it's a source.

Did I miss anything?